src/rabbitmq.js

```
@@ -0,0 +1,356 @@
+/* @flow */
+/* global RabbitMQChannel RabbitMQConnection SubscribeObject */
+'use strict'
+
+const amqplib = require('amqplib')
+const assign = require('101/assign')
+const Immutable = require('immutable')
+const isFunction = require('101/is-function')
+const Promise = require('bluebird')
+
+const logger = require('./logger')
+
+/**
+ * RabbitMQ model. Takes authentication from the following environment
+ * variables (or provided opts):
+ *
+ * - RABBITMQ_HOSTNAME (default: 'localhost')
+ * - RABBITMQ_PORT (default: 5672)
+ * - RABBITMQ_USERNAME (no default)
+ * - RABBITMQ_PASSWORD (no default)
+ *
+ * @private
+ * @author Bryan Kendall
+ * @param {Object} [opts] RabbitMQ connection options.
+ * @param {String} [opts.hostname] Hostname for RabbitMQ.
+ * @param {Number} [opts.port] Port for RabbitMQ.
+ * @param {String} [opts.username] Username for RabbitMQ.
+ * @param {String} [opts.password] Username for Password.
+ */
+class RabbitMQ {
+  channel: RabbitMQChannel;
+  connection: RabbitMQConnection;
+  consuming: Map<string, string>;
+  hostname: string;
+  log: Object;
+  password: string;
+  port: number;
+  subscribed: Set<string>;
+  subscriptions: Map<string, Function>;
+  username: string;
+
+  constructor (opts) {
+    opts = assign({}, opts)
+    this.hostname = opts.hostname ||
+      process.env.RABBITMQ_HOSTNAME ||
+      'localhost'
+    this.port = opts.port || parseInt(process.env.RABBITMQ_PORT, 10) || 5672
+    this.username = opts.username || `${process.env.RABBITMQ_USERNAME}`
+    this.password = opts.password || `${process.env.RABBITMQ_PASSWORD}`
```

you should probably throw if these are not defined or passed in ...